addons/account_edi_ubl_cii/models/ir_actions_report.py

```diff
@@ -0,0 +1,48 @@
+import io
+
+from odoo import models
+from odoo.tools.pdf import OdooPdfFileReader, OdooPdfFileWriter
+
+
+class IrActionsReport(models.Model):
+    _inherit = 'ir.actions.report'
+
+    def _is_invoice_report(self, report_ref):
+        # EXTENDS account
+        # allows to add factur-x.xml to custom PDF templates (comma separated list of template names)
+        custom_templates = self.env['ir.config_parameter'].sudo().get_param('account.custom_templates_facturx_list', '')
+        custom_templates = [report.strip() for report in custom_templates.split(',')]
+        return super()._is_invoice_report(report_ref) or self._get_report(report_ref).report_name in custom_templates
+
+    def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
+        # EXTENDS base
+        collected_streams = super()._render_qweb_pdf_prepare_streams(report_ref, data, res_ids)
+
+        if collected_streams \
+                and res_ids \
+                and len(res_ids) == 1 \
+                and self._is_invoice_report(report_ref) \
+                and not self.env.context.get('from_account_move_send'):  # only triggered from the 'print' action report
```

Comment:
what about this ?
```py
if (
    collected_streams
    and res_ids
    and len(res_ids) == 1
    and self._is_invoice_report(report_ref)
    and not self.env.context.get('from_account_move_send')  # only triggered from the 'print' action report
):
```

Also, we've never embedded factur-x when printing multiple invoices, because the format only supports one invoice. If we would have embedded multiple factur-x.xml inside a single PDF containing multiple invoices, it would most likely not be decoded (at least it wouldn't work in Odoo, I think)

Reply:
Yeah, true that the print joins all documents in one PDF unlike the Send & Print, thanks.